Firebase/Database/Login/FAuthTokenProvider.m

```diff
@@ -16,65 +16,50 @@
 
 #import "FAuthTokenProvider.h"
 #import "FUtilities.h"
+#import "FIRAppInternal.h"
 #import "FIRLogger.h"
 #import "FIRDatabaseQuery_Private.h"
 #import "FIRNoopAuthTokenProvider.h"
 
-static NSString *const FIRAuthStateDidChangeInternalNotification = @"FIRAuthStateDidChangeInternalNotification";
-static NSString *const FIRAuthStateDidChangeInternalNotificationTokenKey = @"FIRAuthStateDidChangeInternalNotificationTokenKey";
-
-
-/**
- * This is a hack that defines all the methods we need from FIRFirebaseApp. At runtime we use reflection to get an
- * actual instance of FIRFirebaseApp. Since protocols don't carry any runtime information and selectors are invoked
- * by name we can write code against this protocol as long as the method signatures of FIRFirebaseApp don't change.
- */
-@protocol FIRFirebaseAppLike <NSObject>
-
-- (void)getTokenForcingRefresh:(BOOL)forceRefresh withCallback:(void (^)(NSString *_Nullable token, NSError *_Nullable error))callback;
-
-@end
-
-
 /**
  * This is a hack that defines all the methods we need from FIRAuth.
  */
 @protocol FIRFirebaseAuthLike <NSObject>
 
-- (id<FIRFirebaseAppLike>) app;
+- (FIRApp *) app;
 
 @end
 
 /**
  * This is a hack that copies the definitions of Firebase Auth error codes. If the error codes change in the original code, this
  * will break at runtime due to undefined behavior!
  */
-typedef NS_ENUM(NSUInteger, FIRErrorCode) {
-    /*! @var FIRErrorCodeNoAuth
+typedef NS_ENUM(NSUInteger, FIRMockErrorCode) {
```

Comment:
I'll take a look - I wasn't actually sure what this mapped to.  There was already [`FIRErrorCode`](https://github.com/firebase/firebase-ios-sdk/blob/31a2acb7497eaaeae38b7ceb17edcab09d7f2866/Firebase/Core/Private/FIRErrorCode.h) declared but doesn't seem to match up with the values here. Do you know where these were copied from?